src/test/ui/parser/issue-65257-invalid-var-decl-recovery.stderr

```diff
@@ -0,0 +1,59 @@
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:2:5
+   |
+LL |     auto n = 0;
+   |     ----^^^^^^
+   |     |
+   |     help: to introduce a variable, write `let` instead of `auto`
+
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:4:5
+   |
+LL |     auto m;
+   |     ----^^
+   |     |
+   |     help: to introduce a variable, write `let` instead of `auto`
+
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:8:5
+   |
+LL |     var n = 0;
+   |     ---^^^^^^
+   |     |
+   |     help: to introduce a variable, write `let` instead of `var`
+
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:10:5
+   |
+LL |     var m;
+   |     ---^^
+   |     |
+   |     help: to introduce a variable, write `let` instead of `var`
+
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:14:5
+   |
+LL |     mut n = 0;
+   |     ---^^^^^^
+   |     |
+   |     help: missing `let`
+
+error: invalid variable declaration
+  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:16:5
+   |
+LL |     mut var;
+   |     ---^^^^
+   |     |
+   |     help: missing `let`
```

Comment:
For cases like this I find that making the suggestion span the entirety of the `let mut name` in `let mut name = foo();` makes it clearer and makes using regular `span_suggestion` worth it:

```
error: invalid variable declaration
  --> $DIR/issue-65257-invalid-var-decl-recovery.rs:16:5
   |
LL |     mut var = 42;
   |     ^^^^^^^ help: missing keyword: `let mut var`
```

Reply:
Including `name` is imo too complicated impl-wise to be worth it and would require special casing, but I've tweaked it a bit.